src/trezor/ui/num_pad.py

```diff
@@ -0,0 +1,48 @@
+from trezor import ui
+from trezor.ui import display
+from trezor.ui.button import BTN_CLICKED, Button
+
+ITEMS_PER_PAGE = 11
+
+
+def digit_area(i):
+    return ui.grid(i + 3)  # skip the first line
+
+
+class NumPad(ui.Widget):
+    def __init__(self, label: str, start: int, end: int):
+        """
+        Generates a numpad with numbers from `start` to `end` excluding.
+        """
+        display.clear()
```

Comment:
This shouldn't really be needed, `ui.layout` usually takes care of clearing the display.